File: problems/2443-sum-of-number-and-its-reverse/solution.ts
function sumOfNumberAndItsReverse(num: number): boolean {
  if (num === 0) return true;

  for (let i = 1; i < num; i++) {
    let rev = 0,
      n = i;

    while (n > 0) {
      rev = rev * 10 + (n % 10);
      n = Math.floor(n / 10);
    }

    if (i + rev === num) return true;
  }

  return false;
}

export { sumOfNumberAndItsReverse };

File: problems/2443-sum-of-number-and-its-reverse/solution.go
package main

func sumOfNumberAndItsReverse(num int) bool {
	if num == 0 {
		return true
	}

	for i := num - 1; i >= num/2; i-- {
		rev := 0

		for n := i; n > 0; {
			rev = (rev * 10) + (n % 10)
			n = n / 10
		}

		if i+rev == num {
			return true
		}
	}

	return false
}
